tests/src/test/scala/cats/tests/ChainSuite.scala

```diff
@@ -119,6 +119,32 @@ class ChainSuite extends CatsSuite {
     }
   }
 
+  test("knownSize should be consistent with size") {
+    forAll { (cu: Chain[Unit]) =>
+      val expected = cu.size match {
+        case size @ (0L | 1L) => size
+        case _                => -1L
+      }
+
+      assertEquals(cu.knownSize, expected)
+    }
+  }
+
+  test("lengthCompare and sizeCompare should be consistent with length and size") {
+    forAll { (cu: Chain[Unit], diff: Byte) =>
```

Comment:
I think a more direct law is:
```scala
signum(cu.lengthCompare(x)) == signum((cu.length.compareTo(x)))
```

Can we test that one?

Reply:
fixed